src/Microsoft.Diagnostics.Monitoring.EventPipe/Triggers/AspNet/AspNetRequestStatusTriggerSettings.cs

```diff
@@ -7,36 +7,64 @@
 using System.ComponentModel.DataAnnotations;
 using System.Linq;
 using System.Text;
-using System.Text.RegularExpressions;
 
 namespace Microsoft.Diagnostics.Monitoring.EventPipe.Triggers.AspNet
 {
     internal sealed class AspNetRequestStatusTriggerSettings : AspNetTriggerSettings
     {
-        private const string StatusRegex = "[1-5][0-9]{2}";
-        private static readonly string RangeRegex = FormattableString.Invariant($"^{StatusRegex}(-{StatusRegex})?$");
-
         /// <summary>
-        /// Specifies the set of status codes for the trigger. This can be individual codes or ranges.
-        /// E.g. 200;400-500
+        /// Specifies the set of status codes for the trigger.
+        /// E.g. 200-200;400-500
         /// </summary>
         [Required]
-        public string[] StatusCodes { get; set; }
+        [CustomValidation(typeof(StatusCodeRangeValidator), nameof(StatusCodeRangeValidator.ValidateStatusCodes))]
+        public StatusCodeRange[] StatusCodes { get; set; }
```

Comment:
Also add MinLength(1) attribute. The custom validator below doesn't need to check that.

Reply:
good point